src/stratigraphy/layer/layer.py

```diff
@@ -41,118 +39,69 @@ def to_json(self) -> dict:
-                    material_description=material_prediction, depth_interval=depth_interval_prediction
+                if start_data is not None
+                else None
+            )
+            end = (
+                DepthColumnEntry(
+                    value=end_data["value"],
+                    rect=fitz.Rect(end_data["rect"]),
+                    page_number=end_data["page"],
                 )
-            else:
-                layer_predictions = Layer(material_description=material_prediction, depth_interval=None)
-
-            page_layer_predictions_list.append(layer_predictions)
-
-        return page_layer_predictions_list
-
-
-def _create_textblock_object(lines: list[dict]) -> TextBlock:
-    """Creates a TextBlock object from a dictionary.
-
-    Args:
-        lines (list[dict]): A list of dictionaries representing the lines.
-
-    Returns:
-        TextBlock: The object.
-    """
-    lines = [TextLine([TextWord(**line)]) for line in lines]
-    return TextBlock(lines)
-
+                if end_data is not None
+                else None
+            )
 
-# TODO: convert to FeatureOnPage[Layer] (cf. ticket LGVISIUM-79)
-@dataclass
-class LayersOnPage:
-    """A class to represent predictions for a single page."""
-
-    layers_on_page: list[Layer]
+            depth_interval_prediction = BoundaryInterval(start=start, end=end)
+        else:
+            depth_interval_prediction = None
 
-    def remove_empty_predictions(self) -> None:
-        """Remove empty predictions from the layers on the page."""
-        self.layers_on_page = [
-            layer for layer in self.layers_on_page if parse_text(layer.material_description.text) != ""
-        ]
+        return Layer(material_description=material_prediction, depth_interval=depth_interval_prediction)
 
 
 @dataclass
 class LayersInDocument:
     """A class to represent predictions for a single document."""
 
-    layers_in_document: list[LayersOnPage]
+    layers: list[Layer]
     filename: str
 
-    def add_layers_on_page(self, layers_on_page: LayersOnPage):
-        """Add layers on a page to the layers in the document.
-
-        Args:
-            layers_on_page (LayersOnPage): The layers on a page to add.
-        """
-        self.layers_in_document.append(layers_on_page)
 
-    def get_all_layers(self) -> list[Layer]:
-        """Get all layers in the document.
+@dataclass
+class IntervalBlockGroup:
+    """A class to represent a group of depth intervals and a group of associated text blocks."""
 
-        Returns:
-            list[Layer]: All layers in the document.
-        """
-        all_layers = []
-        for layers_on_page in self.layers_in_document:
-            all_layers.extend(layers_on_page.layers_on_page)
-        return all_layers
+    depth_intervals: list[Interval]
+    blocks: list[TextBlock]
 
 
 @dataclass
-class IntervalBlockGroup:
-    """A class to represent a group of depth interval blocks."""
+class IntervalBlockPair:
```

Comment:
why do we make a distinction between IntervalBlockGroup and InveralBlockPair?

Reply:
I added a commit with better docstrings for these classes.

I also moved IntervalBlockGroup to the depthcolumn.py file, since it is only used in that file as a helper class.